src/pydocstyle/config.py

```diff
@@ -328,21 +339,17 @@ def _create_check_config(cls, options, use_dafaults=True):
         set for the checked codes.
 
         """
-        match = cls.DEFAULT_MATCH_RE \
-            if options.match is None and use_dafaults \
-            else options.match
-
-        match_dir = cls.DEFAULT_MATCH_DIR_RE \
-            if options.match_dir is None and use_dafaults \
-            else options.match_dir
-
         checked_codes = None
 
         if cls._has_exclusive_option(options) or use_dafaults:
             checked_codes = cls._get_checked_errors(options)
 
-        return CheckConfiguration(checked_codes=checked_codes,
-                                  match=match, match_dir=match_dir)
+        kwargs = dict(checked_codes=checked_codes)
+        for key in ('match', 'match_dir', 'ignore_decorators'):
+            kwargs[key] = getattr(cls, 'DEFAULT_{0}_RE'.format(key.upper())) \
+                if getattr(options, key) is None and use_dafaults \
```

Comment:
I figured that was done intentionally previously for some reason. Happy to change it though.